File: string/1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence/1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence.py

class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        for idx, w in enumerate(sentence.split(' ')):
            if w.startswith(searchWord):
                return idx + 1
        else:
            return -1
            

sentence = "i love eating burger"
searchWord = "burg"
res = Solution().isPrefixOfWord(sentence, searchWord)
print(res)

File: string/1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence/1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence.go
package main

import (
	"fmt"
	"strings"
)

func isPrefixOfWord(sentence string, searchWord string) int {
	words := strings.Split(sentence, " ")
	for i := 0; i < len(words); i++ {
		if strings.HasPrefix(words[i], searchWord) {
			return i + 1
		}
	}
	return -1
}

func main() {
	sentence := "i love eating burger"
	searchWord := "burg"
	res := isPrefixOfWord(sentence, searchWord)
	fmt.Println(res)
}
